pkg/orchestration/wiring/wiringutil/knownshapes/bindable_environment_variables.go

```diff
@@ -1,19 +1,41 @@
 package knownshapes
 
-import "github.com/atlassian/voyager/pkg/orchestration/wiring/wiringplugin"
+import (
+	smith_v1 "github.com/atlassian/smith/pkg/apis/smith/v1"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringplugin"
+)
 
 const (
 	BindableEnvironmentVariablesShape wiringplugin.ShapeName = "voyager.atl-paas.net/BindableEnvironmentVariables"
 )
 
+type ServiceInstanceReference interface {
```

Comment:
This is not needed =) You only need this because you've made the type private. It shouldn't be for this very reason - you cannot use it from another package. But that is the point - to be able to import the type and use it for type safety. Assuming you are writing Go, of course.

Reply:
Also, this is Go anti pattern. Consuming side should be defining interfaces typically, not the producing side. It's not Java =)

Reply:
Good point for the latter comment. Bad habits 🤣 

As for why we made type private - because ShapeName is constant, we didn't want to allow to instantiate struct with arbitrary shape name. Alternatively we discussed changing Contract to contain a map of name -> Shape instead of array of Shapes.

I like constructor approach because ATM we use only some small parts of all available properties, so it makes it easier to instantiate it. But I have no strong opinion about this particular thing, I'm happy to change if we decide to use map.